frontend/src/components/lineage/types.ts

```diff
@@ -0,0 +1,88 @@
+import { PipelineNodeModel, EdgeStyle } from '@patternfly/react-topology';
+import { LineageEntityType } from './LineageNode';
+
+export interface LineageNode {
+  id: string;
+  label: string;
+  entityType: LineageEntityType;
+  features?: number;
+  description?: string;
+  truncateLength?: number;
+  layer?: number; // Optional layer for positioning (0=leftmost, higher=rightward)
+}
+
+export interface LineageEdge {
+  id: string;
+  source: string;
+  target: string;
+  edgeStyle?: EdgeStyle;
+  type?: string;
+  isPositioningEdge?: boolean; // Marks edges used only for layout positioning
+}
+
+export interface LineageData {
+  nodes: LineageNode[];
+  edges: LineageEdge[];
+}
+
+export interface LineageProps {
+  data: LineageData;
+  height?: string;
+  loading?: boolean;
+  error?: string;
+  emptyStateMessage?: string;
+  onNodeSelect?: (nodeId: string | null) => void;
+  className?: string;
+  title?: string;
+}
+
+export const convertToLineageNodeModel = (node: LineageNode): PipelineNodeModel => {
```

Comment:
```suggestion
export const convertToLineageNodeModel = (node: LineageNode): NodeModel => {
```

You aren't using `runAfterTasks` so no need to specify as a pipeline.